### PR TITLE
[Tmi.js] Add type definitions

### PR DESCRIPTION
**Pull request by DyingAlbatross**
Adds definitions for tmi.js

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.

### CODE REVIEW

**Comment by mhegazy** [comment]
https://github.com/DyingAlbatross Can u please address the code failures.

File: types/tmi.js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+  new (options?: TmiOptions): Client;
+}
+
+interface ApiOptions {
+  url: string;
+  headers?: any;
+  method?: string;
+}
+
+interface Client {
```
Comment (leonard-thieu) [inline types/tmi.js/index.d.ts]: `client` is a constructor. You should make this into a class.

**Note**: lowercase c in `client`.

File: types/tmi.js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    getUsername(): string;
+    isMod( channel: string, 
+           username: string
+         ): boolean;
+    readyState(): string;
+
+    /* Events */
+    on( event: "action", 
+        callback: (channel: string, userstate: UserState, message: string, self: boolean) => void
+      ): void;
```
Comment (leonard-thieu) [inline types/tmi.js/index.d.ts]: `on()` should return `this`. It's the `EventEmitter` pattern. `client` actually inherits `EventEmitter` (not the Node one but a custom implementation).

File: types/tmi.js/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,223 @@
+// Type definitions for tmi.js 1.2.1
+// Project: https://github.com/tmijs/tmi.js
+// Definitions by: DyingAlbatross <https://github.com/dyingalbatross>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module 'tmi.js' {
```
Comment (mhegazy) [inline types/tmi.js/index.d.ts]: do not use `declare module`.

**Comment by yuit** [comment]
Thanks for your contribution. This PR has failing CI tests/unaddressed comments and cannot be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**